### ISSUE TITLE
option to limit context shown per line: match +- K columns)

### ISSUE THREAD

**Issue by timotheecour** [BurntSushi/ripgrep#365]
@BurntSushi

sometimes rg shows lines and lines of wrapped text corresponding to a file:line match, this can occur eg for minified js files, html, or json files that are just 1 big line(s) of text.

This will obfuscate the relevant matches, will render slowly (and is even worse if, say, the command is run from a vertically split tmux pane because of the extra overhead in wrapping long lines: in that case also ^C or ^\ doesn't kill the rg process until much later).

of course a workaround is to exclude these files via `-g` but oftentimes this is undesirable and can be tedious depending on codebase (extension based filtering isn't good enough in my use cases)

Long story short: could we have:

```
--after-context-columns
--before-context-columns
--context-columns
```
which is analog of 

```
--after-context
--before-context
--context
```
but for horizontal, instead of vertical, context

Another useful option:
`--skip-line-greater-than=N`
which would skip a line greater than specified N columns or bytes

**Comment by BurntSushi** [comment]
Dupe of #129

**Comment by kdar** [comment]
Should we reopen this issue since the first part of it is still relevant and not fixed by --max-columns?

**Comment by abitrolly** [comment]
Yes, I don't see how to extend `--only-matching` to include more context around.

**Comment by PanieriLorenzo** [comment]
please reopen this, it is not a duplicate of #129
